pkg/utils/map.go

```diff
@@ -34,7 +36,7 @@ func MergeMapNested(allowOverwrite bool, sources ...map[string]interface{}) (map
 						} else {
 							destination[k] = merged
 						}
-					} else if allowOverwrite {
+					} else if allowOverwrite || destVal == nil {
```

Comment:
This small modification in `MergeMapNested` and `PutMapNested` was motivated by the fact that existing map entries with nil values were treated like non-nil ones, which is not what we want when merging or updating configs.